client/src/pages/Feed.js

```diff
@@ -416,6 +416,54 @@ const Feed = (props) => {
     }
   }, [location, page, filterType, selectedOptions, applyFilters]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  useEffect(() => {
+    // Onboarding
+    if (props.history.location.state) {
+      const handleOnboardingOptions = (option, label) => {
+        optionsDispatch({ type: ADD_OPTION, payload: { option, label } });
+      };
+
+      const {
+        postType,
+        helpType,
+        location,
+        providers,
+      } = props.history.location.state;
+      location && dispatchAction(SET_VALUE, "location", location);
+      if (postType === "Requesting help") {
+        // requesting help
+
+        handleChangeType({ key: "REQUEST" });
+        if (helpType === "medical") {
+          let option = filters[2].options[0];
+          handleOnboardingOptions(option, "type");
+        } else {
+          for (let i = 1; i < filters[2].options.length; ++i) {
+            let option = filters[2].options[i];
+            handleOnboardingOptions(option, "type");
+          }
+        }
+      } else {
+        // offering help
+        handleChangeType({ key: "OFFER" });
+        if (providers) {
+          let organizationFilter = providers.filter(
+            (option) => option === "As an Organisation",
+          );
+          if (organizationFilter.length > 0) {
+            for (let i = 1; i < filters[1].options.length; ++i) {
+              let option = filters[1].options[i];
+              handleOnboardingOptions(option, "providers");
+            }
+          } else {
+            let option = filters[1].options[0];
+            handleOnboardingOptions(option, "providers");
+          }
+        }
+      }
+    }
+  }, [filters, handleChangeType, props.history.location.state]);
```

Comment:
I have no Idea why it keeps filling in these fields in this array when I commit I hope someone can let me know why this is the case.

Reply:
Will take a look and fix if neeeded

Reply:
How should it be like?

Reply:
it should be like     }, []);

Reply:
Cool, I have resolved this cuz I want to merge this one asap